### PR TITLE
Complete transfrom coverage in RS e2e tests

### PR DESCRIPTION
**Pull request by jbiskie**
# Description

Adds RS e2e test for `ucsdOruRemoveAccessionNumberObservation`
Updates the description of `ucsdOruUpdateUniversalServiceIdentifier` transform to correspond to current usage
Updates the RS e2e test criteria for `UCSD ORU requirements` to use `input.ORC-12` instead of `ORC-12`

## Issue

#1539

## Testing instructions

Run RS e2e tests via Github Actions "Automated Staging Test - Run integration tests". Ensure all tests pass.

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: rs-e2e/src/main/resources/assertion_definitions.json [hunk 1]
```diff
@@ -91,6 +91,15 @@
       "rules": [
         "OBX-6 = 'µmol/L'"
       ]
+    },
+    {
+      "name": "UCSD ORU remove OBX when OBX-3.4 is 99717-5",
```
[the remark this answers is not in the document]
Comment (jbiskie) [reply rs-e2e/src/main/resources/assertion_definitions.json:96]: This test is conditioned to run against a single input file only

File: rs-e2e/src/main/resources/assertion_definitions.json [hunk 2]
```diff
@@ -91,6 +91,15 @@
...
+      "conditions": [
+        "MSH-10 = '009'"
+      ],
+      "rules": [
+        "OBX.count() = 1"
```
Comment (GilmoreA6) [inline rs-e2e/src/main/resources/assertion_definitions.json:101]: I would add an additional rule that checks that OBX-3 = "57721-3" so we know that the remaining obx is the one we expect

**Review by GilmoreA6 — CHANGES_REQUESTED** [review]
One suggestion, other than that looks good. Thanks for cleaning up those two items.

**Review by GilmoreA6 — APPROVED** [review]

File: rs-e2e/src/main/resources/assertion_definitions.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
         "ORC-21.10 = MSH-6",
         "OBR-2.1 = ORC-2.1",
         "OBR-2.2 = ORC-2.2",
-        "OBR-16 = ORC-12",
+        "OBR-16 = input.ORC-12",
```
[the remark this answers is not in the document]
Comment (jbiskie) [reply rs-e2e/src/main/resources/assertion_definitions.json:36]: Correct, makes no difference if ORC-12 is unchanged (and it should not be). Using input.ORC-12 will catch if an inadvertent change was made to ORC-12.

File: rs-e2e/src/main/resources/assertion_definitions.json (hunk 2 above)
Comment (jbiskie) [reply rs-e2e/src/main/resources/assertion_definitions.json:102]: Done

**Pull request merged**